src/main/kotlin/com/autonomousapps/advice/Advice.kt

```diff
@@ -87,13 +92,23 @@ data class Advice(
   fun isAdd() = fromConfiguration == null && !isCompileOnly()
 
   /**
-   * An advice is "remove-advice" if it is declared and not used, AND is not `compileOnly`.
+   * An advice is "remove-advice" if it is declared and not used, AND is not `compileOnly`,
+   * AND is not `processor`.
    */
-  fun isRemove() = toConfiguration == null && !isCompileOnly()
+  fun isRemove() = toConfiguration == null && !isCompileOnly() &&!isProcessor()
```

Comment:
This is fine. I think you're right and we need some easy way to distinguish dependency-related remove advice from processor-related remove advice. Having an `isProcessor()` method makes sense.